src/main/java/net/rptools/maptool/client/functions/RESTfulFunctions.java

```diff
@@ -269,9 +270,15 @@ private boolean isFullResponseRequested(List<Object> parameters) {
         || (headerIndex == parameters.size() - 1 && isLastParamBoolean(parameters))) {
       return new HashMap<String, List<String>>();
     } else {
-      return gson.fromJson(
-          (String) parameters.get(headerIndex),
-          new TypeToken<Map<String, List<String>>>() {}.getType());
+      String headerString;
+      Object paramHeader = parameters.get(headerIndex);
+      if (paramHeader instanceof JsonObject) {
+        headerString = ((JsonObject) paramHeader).toString();
+      } else {
+        headerString = paramHeader.toString();
+      }
```

Comment:
I believe this can be simplified to just `String headerString = parameters.get(headerIndex).toString()`.

Reply:
Doh!   Yeah, that's better.